Better Error handling for Invalid Auth Header

If the auth header is not empty but invalid the program will panic. This is bad

```shell
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
main.main()
        /home/gianni/go/src/hardcover-quotes/cmd/hcq/main.go:140 +0x7cf
```

--- a/cmd/hcq/main.go
+++ b/cmd/hcq/main.go
@@ -135,7 +135,10 @@ func main() {
 	client := graphql.NewClient(apiURL)
 	ctx := context.Background()
 
-	user_info_response, _ := queryUserInfo(ctx, *client, authToken)
+	user_info_response, err := queryUserInfo(ctx, *client, authToken)
+	if err != nil {
+		log.Fatal("query error", err)
+	}
 
 	quotedBooks := user_info_response.Me[0].Quoted_books
 

